### ISSUE TITLE
DOIs with resource type dataset not showing up in Google Dataset search

### ISSUE THREAD

**Issue by MaryHirsch** [datacite/datacite#1119]
### Expected Behaviour

DataCite DOis with resourceType Dataset can be indexed in Google dataset search

### Current Behaviour

Some DOIs are not appearing even when resource type "dataset" in schema.org

### Steps to Reproduce

examples:
https://api.datacite.org/dois/application/ld+json/10.25611/rryf-x193 
https://api.datacite.org/dois/application/ld+json/10.25611/dyma-2n05

Full list provided by Rice:
https://doi.org/10.25611/rryf-x193
https://doi.org/10.25611/dyma-2n05
https://doi.org/10.25611/10pr-5804
https://doi.org/10.25611/ayzm-0962
https://doi.org/10.25611/kytg-mw69

### Context (Environment)

Production

### Possible Implementation

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2o5hx)

**Comment by mfenner** [comment]
@kjgarza please confirm that we are correctly exposing the schema.org metadata for those dois in DataCite Search.

**Comment by kjgarza** [comment]
One can check the Search using the **google structured-data testing tool** https://search.google.com/structured-data/testing-tool/u/0/ just copy paste the links from DC Search.

https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fsearch.datacite.org%2Fworks%2F10.25611%2Frryf-x193

https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fsearch.datacite.org%2Fworks%2F10.25611%2Fdyma-2n05

https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fsearch.datacite.org%2Fworks%2F10.25611%2F10pr-5804

https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fsearch.datacite.org%2Fworks%2F10.25611%2Fayzm-0962

https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fsearch.datacite.org%2Fworks%2F10.25611%2Fkytg-mw69

All of them have the DOI metadata in schema.org format in the page and they are type dataset. In essence they should be indexed.

**Comment by mfenner** [comment]
Thanks @kjgarza. Last thing we can do is check that Google is processing the sitemaps files correctly. I can check that.

**Comment by KellyStathis** [comment]
With https://github.com/datacite/datacite/issues/1328 fixed I am working on this now, waiting for indexing to complete.

**Comment by KellyStathis** [comment]
Indexing is taking some time.

The Google Dataset Search team has suggested that we publish a sitemap with only datasets to speed up indexing: https://app.frontapp.com/open/cnv_aw1m5xx?key=jGhyzWxmwGcpbNyUE8wo7xunYMeosEuF

I suspect this is more of an issue now than it was ~3 years ago because they are no longer the majority of DOIs. With our current sitemap index including all DOIs, Google is having to crawl lots of non-dataset pages, only to discover they aren't datasets and decide not to index them. From my understanding, if they are crawling lots of pages that aren't indexed this will lower our priority and slow our crawl rate.

@sarala @richardhallett I realize this is getting into feature request territory. Would it be possible to limit the sitemap to DOIs with resourceTypeGeneral "Dataset" only? Since Google Dataset Search is the main purpose of the Commons sitemap and we don't recommend it as an option for harvesting, I don't see this having an impact on other users.

**Comment by richardhallett** [comment]
The sitemaps exist as overall view of all our data, the fact it's generated for commons now means it's an index of all the data in commons, which obviously includes non datasets.

I'm not sure it makes sense for us to be providing limited sitemaps, as this would then be very specific for Google, other harvesters may want the total sitemap.
I'm not sure if there is scope for providing an alternate sitemap that is DataSet only, is it possible for Google to accept a different sitemap root file? i.e. commons.datacite.org/sitemap-datasets.xml.gz or something, if that was the case we might be able to do it, but it really is a new feature, I'm not sure of development effort but certainly something that'd need planning CC: @sarala 

Overall I suspect part of the problem is Commons is not overly quick, so parsing any metadata to filter will be a challenge. I'm not sure how they are actually looking at metadata, if it's via our APIs or page parsing?

**Comment by KellyStathis** [comment]
Yes, if we had a second datasets-only sitemap index we could provide that to Google i.e. commons.datacite.org/sitemap-datasets.xml.gz - we specify the location in Google Search Console. 

I like the idea to have two so that no existing functionality is broken. That said we don't recommend or document sitemaps for harvesting anywhere, so I would be surprised if many harvesters are using it.

Google is retrieving and parsing the schema.org metadata embedded in each Commons landing page visited by the crawler, not using our APIs.

**Comment by codycooperross** [comment]
There have been many conversations about this, and I believe we've arrived at Google Dataset Search indexing as a product improvement given the challenges and constraints outlined here and elsewhere. Capturing the conversation here: https://github.com/datacite/product-backlog/issues/290
